pkg/apis/core/validation/validation.go

```diff
@@ -4356,10 +4357,24 @@ func ValidateConfigMap(cfg *core.ConfigMap) field.ErrorList {
 		for _, msg := range validation.IsConfigMapKey(key) {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("data").Key(key), key, msg))
 		}
+		// check if the value stored consists entirely of valid UTF-8-encoded runes
+		if !utf8.ValidString(cfg.Data[key]) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("data").Key(key), key, "invalid byte sequence in UTF-8"))
+		}
+		// check if we have a duplicate key in the other bag
+		if _, isValue := cfg.BinaryData[key]; isValue {
+			allErrs = append(allErrs, field.Duplicate(field.NewPath("data").Key(key), "duplicate of key in binaryData"))
```

Comment:
How does this come through?  The second arg is supposed to be the duplicate value, but Duplicate() may be the wrong error here.  Or it may be worth adding an extra param to Duplicate.  Or maybe we need a new error function?

Reply:
I'd probably go with `field.Invalid` and explain in the message

Reply:
gotcha @liggitt doing that now